src/Storages/RocksDB/StorageEmbeddedRocksDB.cpp

```diff
@@ -166,14 +166,24 @@ StorageEmbeddedRocksDB::StorageEmbeddedRocksDB(const StorageID & table_id_,
         bool attach,
         ContextPtr context_,
         const String & primary_key_,
-        Int32 ttl_)
+        Int32 ttl_,
+        String rocksdb_dir_,
+        bool read_only_)
     : IStorage(table_id_)
     , WithContext(context_->getGlobalContext())
     , primary_key{primary_key_}
     , ttl(ttl_)
+    , read_only(read_only_)
 {
     setInMemoryMetadata(metadata_);
-    rocksdb_dir = context_->getPath() + relative_data_path_;
+    if (rocksdb_dir_.empty())
+    {
+        rocksdb_dir = context_->getPath() + relative_data_path_;
+    }
+    else
+    {
+        rocksdb_dir = rocksdb_dir_;
+    }
```

Comment:
```suggestion
    if (rocksdb_dir_.empty())
        rocksdb_dir = context_->getPath() + relative_data_path_;
```

Reply:
we need to pass `rocksdb_dir_` to `rocksdb_dir` when `rocksdb_dir_` is not empty.